library/src/main/java/com/northconcepts/templatemaster/form/CrudResource.java

```diff
@@ -337,7 +338,25 @@ protected Response postEditRecordImpl(ID id, UriInfo uriInfo, ENTITY form) throw
         return gotoPath(subUrl + "/view/" + getId(form));
     }
     
+    //==========================================================================================
+    // Clone
+    //==========================================================================================
+    
+    @GET
+    @Path("/clone/{id}")
+    public Response getCloneRecord(@PathParam("id") ID id) throws Throwable {
+        if (!formDef.isAllowClone()) {
+            setErrorFlashMessage("Clone not allowed");
+            return badRequest();
```

Comment:
@gvjoshi25 does it make sense to overload the badRequest method to take an error message?

Reply:
We have to display that error message to `error-400.html`, right?

Reply:
Yes, I believe so.

Reply:
As discussed, we will implement that later. Hence, resolving this conversion